File: leetcode/167. Two Sum 2 Input array is sorted/167_two_sum_2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers)-1
        answer = []
        while left < right:
            if numbers[left] + numbers[right] == target:
                answer.append(left+1)
                answer.append(right+1)
                break
            
            elif target-numbers[left] < numbers[right]:
                right -= 1
            else:
                left += 1
        return answer

File: leetcode/167. Two Sum 2 Input array is sorted/167_two_sum_2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        map<int,int> num_map;
        vector<int> answer(2);
        
        for(int i=0; i<numbers.size(); i++){
            int map_value = target - numbers[i];
            if(num_map.find(map_value) != num_map.end())
            {
                answer[0] = num_map[map_value]+1;
                answer[1] = i+1;
                return answer;
            }
            num_map.insert(make_pair(numbers[i], i));
        }
        return answer;
    }
};
